// dom-data-bind/src/ComponentManager.js
/**
 * @file 组件管理
 * @author yibuyisheng(yibuyisheng@163.com)
 */

function ComponentManager() {
    this.components = {};
}

ComponentManager.prototype.registe = function (ComponentClass) {
    // var name = getClassName(ComponentClass);
    var name = ComponentClass.name;
    this.components[name] = ComponentClass;
    this.mountStyle(name);
};

ComponentManager.prototype.getClass = function (name) {
    return this.components[name];
};

/**
 * 将组件的样式挂载上去
 *
 * @private
 * @param {string} componentClassName 组件js类名
 */
ComponentManager.prototype.mountStyle = function (componentClassName) {
    var style = this.components[componentClassName].getStyle();

    var styleNode = document.createElement('style');
    styleNode.innerHTML = style.replace(
        /#root#/g,
        '.' + this.components[componentClassName].getComponentClassName()
    );
    document.head.appendChild(styleNode);
};


module.exports = ComponentManager;

function getClassName(klass) {
    return klass.toString().match(/^function\s*(\w+?)(?=\(\w*?\))/)[1];
}
